flopy/modflow/mfflwob.py

```diff
@@ -314,3 +350,230 @@ def write_file(self):
+
+        # read dataset 2 -- TOMULTFB
+        line = f.readline()
+        t = line.strip().split()
+        tomultfb = float(t[0])
+
+        nqobfb = np.zeros(nqfb, dtype=np.int32)
+        nqclfb = np.zeros(nqfb, dtype=np.int32)
+        obsnam = []
+        irefsp = []
+        toffset = []
+        flwobs = []
+
+        layer = []
+        row = []
+        column = []
+        factor = []
+
+        # read datasets 3, 4, and 5 for each of nqfb groups
+        # of cells
+        nobs = 0
+        while True:
+
+            # read dataset 3 -- NQOBFB NQCLFB
+            line = f.readline()
+            t = line.strip().split()
+            nqobfb[nobs] = int(t[0])
+            nqclfb[nobs] = int(t[1])
+
+            # read dataset 4 -- OBSNAM IREFSP TOFFSET FLWOBS
+            ntimes = 0
+            while True:
+                line = f.readline()
+                t = line.strip().split()
+                obsnam.append(t[0])
+                irefsp.append(int(t[1]))
+                toffset.append(float(t[2]))
+                flwobs.append(float(t[3]))
+                ntimes += 1
+                if ntimes == nqobfb[nobs]:
+                    break
+
+            # read dataset 5 -- Layer Row Column Factor
+            k = np.zeros(abs(nqclfb[nobs]), np.int32)
+            i = np.zeros(abs(nqclfb[nobs]), np.int32)
+            j = np.zeros(abs(nqclfb[nobs]), np.int32)
+            fac = np.zeros(abs(nqclfb[nobs]), np.float32)
+
+            ncells = 0
+            while True:
+                line = f.readline()
+                t = line.strip().split()
+                k[ncells] = int(t[0])
+                i[ncells] = int(t[1])
+                j[ncells] = int(t[2])
+                fac[ncells] = float(t[3])
+
+                ncells += 1
+                if ncells == abs(nqclfb[nobs]):
+                    layer.append(k)
+                    row.append(i)
+                    column.append(j)
+                    factor.append(fac)
+                    break
+
+            nobs += 1
+            if nobs == nqfb:
+                break
+
+        irefsp = np.array(irefsp) - 1
+        layer = np.array(layer) - 1
+        row = np.array(row) - 1
+        column = np.array(column) - 1
+        factor = np.array(factor)
+
+        # close the file
+        f.close()
+
+        # get ext_unit_dict if none passed
+        if ext_unit_dict is None:
+            namefile = os.path.join(model.model_ws, model.namefile)
+            ext_unit_dict = parsenamefile(namefile, model.mfnam_packages)
+
+        flowtype, ftype = _get_ftype_from_filename(f.name, ext_unit_dict)
+
+        # set package unit number
+        unitnumber = None
+        filenames = [None, None]
+        if ext_unit_dict is not None:
+            unitnumber, filenames[0] = \
+                model.get_ext_dict_attr(ext_unit_dict,
+                                        filetype=ftype.upper())
+            if iufbobsv > 0:
+                _, filenames[1] = \

```

Comment:
I would use `key` instead of `_` just so it is clear what the first value is.

Reply:
Reverted commit eca305ab to address this one.